collagraph/component.py

```diff
@@ -43,3 +60,9 @@ def before_unmount(self):
     @abstractmethod
     def render():  # pragma: no cover
         pass
+
+    def __del__(self):
+        # Make sure to clean-up _element. Normally this shouldn't be needed as it is
+        # a weak ref, but if a type of element was set for which no weak ref can be
+        # made, we still need to do the clean-up.
+        self._element = None
```

Comment:
Then why even bother with a weakref?

Reply:
Because this problem was an afterthought :D Yeah, I guess the weakref ain't needed if I just keep this cleanup step.